lib/explorer/data_frame.ex

```diff
@@ -2973,6 +2973,126 @@ defmodule Explorer.DataFrame do
+  Providing no options will relocate the columns to beginning of the DataFrame.
+
+  ## Options
+
+    * `:before` - Specifies to relocate before the given column. You can also pass `:first` and `:last` to relocate to the beginning or end of the DataFrame.
+
+    * `:after` - Specifies to relocate after the given column. You can also pass `:first` and `:last` to relocate to the beginning or end of the DataFrame.
+
+  ## Examples
+
+  Relocate a single column
+
+      iex> df = Explorer.DataFrame.new(a: ["a", "b", "a"], b: [1, 3, 1], c: [nil, 5, 6])
+      iex> Explorer.DataFrame.relocate(df, "a", after: "c")
+      #Explorer.DataFrame<
+        Polars[3 x 3]
+        b integer [1, 3, 1]
+        c integer [nil, 5, 6]
+        a string ["a", "b", "a"]
+      >
+
+  Relocate (and reorder) multiple columns to the beginning
+
+      iex> df = Explorer.DataFrame.new(a: [1, 2], b: [5.1, 5.2], c: [4, 5], d: ["yes", "no"])
+      iex> Explorer.DataFrame.relocate(df, ["d", 1], before: :first)
+      #Explorer.DataFrame<
+        Polars[2 x 4]
+        d string ["yes", "no"]
+        b float [5.1, 5.2]
+        a integer [1, 2]
+        c integer [4, 5]
+      >
+
+  Relocate before another column
+
+      iex> df = Explorer.DataFrame.new(a: [1, 2], b: [5.1, 5.2], c: [4, 5], d: ["yes", "no"])
+      iex> Explorer.DataFrame.relocate(df, ["a", "c"], before: "b")
+      #Explorer.DataFrame<
+        Polars[2 x 4]
+        a integer [1, 2]
+        c integer [4, 5]
+        b float [5.1, 5.2]
+        d string ["yes", "no"]
+      >
+  """
+
+  @doc type: :single
+  @spec relocate(
+          df :: DataFrame.t(),
+          columns :: [column()] | column(),
+          opts :: Keyword.t()
+        ) :: DataFrame.t()
+
+  def relocate(df, columns_or_column, opts)
+
+  def relocate(df, column, opts) when is_column(column),
+    do: relocate(df, [column], opts)
+
+  def relocate(df, columns, opts) do
+    opts = Keyword.validate!(opts, before: nil, after: nil)
+
+    columns = to_existing_columns(df, columns)
+
+    {new_names, col_index} =
+      case {opts[:before], opts[:after]} do
+        {nil, nil} ->
+          {:before, :first}
+
+        {before_col, nil} ->
+          {:before, before_col}
+
+        {nil, after_col} ->
+          {:after, after_col}
+
+        {before_col, after_col} ->
+          raise(
+            ArgumentError,
+            "only one location must be given. Got both " <>
+              "before: #{inspect(before_col)} and after: #{inspect(after_col)}"
+          )
+      end
+      |> relocate_columns(df, columns)
+
+    out_df = %{df | names: new_names}
+
+    Shared.apply_impl(df, :relocate, [out_df, columns, col_index])
+  end
+
+  defp relocate_columns({direction, :first}, df, columns_to_relocate),
+    do: relocate_columns({direction, 0}, df, columns_to_relocate)
+
+  defp relocate_columns({direction, :last}, df, columns_to_relocate),
+    do: relocate_columns({direction, -1}, df, columns_to_relocate)
```

Comment:
I suppose this "special casing" is somewhat redundant when 0 and -1 can be provided as the index. Should I omit this in favor for consistency with other functions that don't use `:first` and `:last` in this way?

Reply:
I like the readability of using :first | :last, so I would keep it. :)

Reply:
Actually, there is an ambiguity if you have a column name first or last, so we should probably remove it. :)

Reply:
Yeah that's true. Although you'd be able to specify the column with a string instead. I'm leaning on removing it still, though

Reply:
Removed in 09d2bf718a11ee9d670135a3c62fe4646e25ba8c, will look at the lazy data frames now!